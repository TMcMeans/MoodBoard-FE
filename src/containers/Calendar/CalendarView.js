import React, { Component } from 'react';
import Calendar from 'react-calendar/dist/entry.nostyle';
import { Link } from 'react-router-dom';

import Button from '../../components/Button/Button.js';

class CalendarView extends Component {
  constructor() {
    super();
    this.state = {
      date: new Date()
    };
  }

  onChange = date => {
    this.setState({ date });
  };

  render() {
    return (
      <div>
        <Calendar value={this.state.value} onChange={this.onChange} />
        <Link to="/affirmations"><Button text="View affirmations" /></Link>
      </div>
    );
  }
}

export default CalendarView;
