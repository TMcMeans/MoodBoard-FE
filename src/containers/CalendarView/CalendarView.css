/*COMPONENT*/
.calendar {
  border: none;
  height: 100vh;
}

/*CALENDAR*/
.react-calendar {
  height: 60vh;
  margin: auto;
  margin-top: 2rem;
  width: 95%;
}

/*CALENDAR NAVIGATION*/
.react-calendar__navigation {
  width: 80%;
  margin: auto;
  margin-top: 4rem;
}

/*NAVIGATION ARROWS*/
.react-calendar__navigation__arrow {
  background: #daa520;
  box-shadow: 0px 2px 4px rgba(255, 255, 255, 0.25),
    inset 5px 10px 10px rgba(0, 0, 0, 0.25);
  border: none;
  border-radius: 10%;
  color: #ffffff;
  font-size: 2rem;
  height: 1.5rem;
  line-height: 0px;
  margin: auto;
  padding: 2px;
  padding-bottom: 8px;
  width: 1.5rem;
}

.react-calendar__navigation__prev2-button,
.react-calendar__navigation__next2-button {
  display: none;
}

/*NAVIGATION MONTH*/
.react-calendar__navigation__label {
  font-size: 2rem;
  color: #ffffff;
  font-family: 'Raleway', sans-serif;
}

/*MONTH DAYS*/
.react-calendar__month-view__weekdays {
  color: #2a2a2a;
}

/*MONTH DATES*/
.react-calendar__tile {
  color: #ffffff;
  font-family: 'Raleway', sans-serif;
  font-size: 2rem;
  text-shadow: 2px 2px 2px #DAA520;
}

.react-calendar__month-view__days__day--neighboringMonth {
  color: #2a2a2a;
}
