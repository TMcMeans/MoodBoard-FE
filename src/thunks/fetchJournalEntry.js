import { addJournalEntry, hasErrored } from '../actions';

export const fetchJournalEntry = url => {
  return async dispatch => {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const journal_entry = await response.json();
      dispatch(getJournalEntry(journal_entry));
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
