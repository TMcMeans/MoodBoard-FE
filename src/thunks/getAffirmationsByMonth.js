import { hasErrored } from '../actions';

export const getAffirmationsByMonth = url => {
  return async dispatch => {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const affirmations = await response.json();
      return affirmations;
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
