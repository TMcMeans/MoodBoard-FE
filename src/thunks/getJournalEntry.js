import { saveJournalEntry, hasErrored } from '../actions';

export const getJournalEntry = url => {
      console.log('running')
  return async dispatch => {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const journal_entry = await response.json();
      console.log(journal_entry)
      dispatch(saveJournalEntry(journal_entry));
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
