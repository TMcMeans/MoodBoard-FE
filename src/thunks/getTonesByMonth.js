import { saveMonthTones, hasErrored } from '../actions';

export const getTonesByMonth = (url) => {
  return async dispatch => {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const tone_data = await response.json();
      await dispatch(saveMonthTones(tone_data.data));
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
