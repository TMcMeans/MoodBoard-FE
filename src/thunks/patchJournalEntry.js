import { hasErrored } from '../actions';

export const patchJournalEntry = (url, entry_text) => {
  return async dispatch => {
    try {
      const response = await fetch(url, {
        method: 'PATCH',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          journal_entry: { entry_text }
        })
      });
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
