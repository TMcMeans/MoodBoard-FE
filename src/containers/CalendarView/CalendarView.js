import React, { Component } from 'react';
import Calendar from 'react-calendar/dist/entry.nostyle';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Logo from '../../components/Logo/Logo';
import { getTonesByMonth } from '../../thunks/getTonesByMonth';
import './CalendarView.css';

export class CalendarView extends Component {
  constructor() {
    super();
    this.state = {
      date: new Date()
    };
  }

  onChange = date => {
    this.setState({ date });
  };

  componentDidMount = async () => {
    // get the date (mo, yr)
    console.log(this.state.date);
    // get the tones
    // set the tones to colors
    // assign colors to individual days (by the day's value)
  };

  render() {
    const prevDays = [28, 29, 30, 31];
    const days = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28];
    const nextDays = [1, 2, 3];
    const prevDayBtns = prevDays.map((day) => {
      return (
        <button className="calendar-day-prev">{day}</button>
      )
    });
    const dayBtns = days.map((day) => {
      return (
        <button className="calendar-day">{day}</button>
      )
    });
    const nextDayBtns = nextDays.map((day) => {
      return (
        <button className="calendar-day-next">{day}</button>
      )
    });
        // <Logo />
    return (
      <div className="calendar">
        <div className="calendar-container">
          <div className="calendar-nav-btns">
            <button className="nav-arrow">&#60;</button>
            <button className="nav-label">February 2019</button>
            <button className="nav-arrow">&#62;</button>
          </div>
          <div className="month-container">
            {prevDayBtns}
            {dayBtns}
            {nextDayBtns}
          </div>
        </div>
        <Calendar value={this.state.value} onChange={this.onChange} />
      </div>
    );
  }
        // <Link to="/affirmations">
        //   <button className="standard-btn light affirmations-btn">
        //     your affirmations
        //   </button>
        // </Link>
}

const mapDispatchToProps = dispatch => ({
  getTonesByMonth: url => dispatch(getTonesByMonth(url))
});

CalendarView.propTypes = {
  getTonesByMonth: PropTypes.func,
  date: PropTypes.string
};

export default connect(
  null,
  mapDispatchToProps
)(CalendarView);
