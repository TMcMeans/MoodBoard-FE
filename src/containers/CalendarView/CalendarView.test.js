import React from 'react';
import { shallow } from 'enzyme';
import { CalendarView } from './CalendarView';
import { getTonesByMonth } from '../../thunks/getTonesByMonth';

jest.mock('../../thunks/getTonesByMonth');

describe('CalendarView', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CalendarView tones={['sadness', 'confident', 'anger']} />
    );
  });

  it('should render a snapshot with correct data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set initial state of date when component mounts', () => {});

  it('should change state of date when onChange is called', () => {});

  it('mapDispatchToProps should dispatch getTonesByMonth onComponentDidMount', () => {});
});

// it('should dispatch deleteFavorite thunk when removeFavorite is called from props', () => {
//   deleteFavorite.mockImplementation(() => {});
//   const expected = deleteFavorite(2, 3);
//   mappedProps.removeFavorite(2, 3);

//   expect(mockDispatch).toHaveBeenCalledWith(expected);
// });
