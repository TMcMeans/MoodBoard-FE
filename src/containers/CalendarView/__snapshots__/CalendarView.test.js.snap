// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarView should render a snapshot with correct data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CalendarView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <Calendar
          maxDetail="month"
          minDetail="century"
          onChange={[Function]}
          returnValue="start"
          showNavigation={true}
          showNeighboringMonth={true}
          view="month"
        />,
        <Link
          replace={false}
          to="/affirmations"
        >
          <button
            className="standard-btn light affirmations-btn"
          >
            your affirmations
          </button>
        </Link>,
      ],
      "className": "calendar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "maxDetail": "month",
          "minDetail": "century",
          "onChange": [Function],
          "returnValue": "start",
          "showNavigation": true,
          "showNeighboringMonth": true,
          "value": undefined,
          "view": "month",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="standard-btn light affirmations-btn"
          >
            your affirmations
          </button>,
          "replace": false,
          "to": "/affirmations",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "your affirmations",
            "className": "standard-btn light affirmations-btn",
          },
          "ref": null,
          "rendered": "your affirmations",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <Calendar
            maxDetail="month"
            minDetail="century"
            onChange={[Function]}
            returnValue="start"
            showNavigation={true}
            showNeighboringMonth={true}
            view="month"
          />,
          <Link
            replace={false}
            to="/affirmations"
          >
            <button
              className="standard-btn light affirmations-btn"
            >
              your affirmations
            </button>
          </Link>,
        ],
        "className": "calendar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "maxDetail": "month",
            "minDetail": "century",
            "onChange": [Function],
            "returnValue": "start",
            "showNavigation": true,
            "showNeighboringMonth": true,
            "value": undefined,
            "view": "month",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="standard-btn light affirmations-btn"
            >
              your affirmations
            </button>,
            "replace": false,
            "to": "/affirmations",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "your affirmations",
              "className": "standard-btn light affirmations-btn",
            },
            "ref": null,
            "rendered": "your affirmations",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
